api/exec/common.go

```diff
@@ -2,4 +2,51 @@ package exec
 
 import "regexp"
 
-var stackNameNormalizeRegex = regexp.MustCompile("[^-_a-z0-9]+")
\ No newline at end of file
+var stackNameNormalizeRegex = regexp.MustCompile("[^-_a-z0-9]+")
+
+var _exists = struct{}{}
+
+type StringSet map[string]struct{}
```

Comment:
This is a workaround so it must be removed once docker-compose to be upgraded to the latest version (2.x). Put them together so we can be easy to rermove them.

It can save mem with struct{}.

Reply:
As these sets are temporary and not big, I agree with this answer https://stackoverflow.com/a/37320521/681629 to say that boolean syntax is clearer.

I understand the requirement that they both come together, but at least rename the file to string_set, or smthing.

Reply:
Okay, I will change